### PR TITLE
Mg/create flask web

### LINKED ISSUE #4: Message board

### LINKED ISSUE #15: Create Flask webserver layer

### PR DESCRIPTION
**Pull request by Matej4545**
Initial "prototype" functionality of server. 

## Frontend
Using Flask, now we have frontend running at port 5000. Messages sent via `client.py` are shown here (after update).

## Logging
Probably not related and needed, but good practice and I used it at least for debug (also replaced print statements). Logging is either to console and to file. Probably leaking some data in debug (connection string).

## (De)serialization
Messages and Clients (represented as classes in `interfaces.py`) are loaded from file during initialization and saved when exiting. Currently it is only plaintext file. Plan is to replace this with db at some point, but this was easy enough and it can serve quite well for testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,3 +37,22 @@ When connecting to a TCP TPM, a simulator needs to be running and listening on p
...
+Windows:
+`python -m venv venv`
+
+Linux: `python3 -m venv venv` (untested)
+
+2. start virtual environment
+
+Windows: `./venv/Scripts/activate`
+
+Linux: `./venv/Scripts/activate`
```
Comment (T0mmas) [inline README.md]:
```suggestion
Linux: `source venv/bin/activate`
```

File: README.md [hunk 2]
```diff
@@ -37,3 +37,22 @@ When connecting to a TCP TPM, a simulator needs to be running and listening on p
...
+
+#### Server prerequisities
+To run server, you will need to do following steps:
+
+1. set up venv environment (run in project directory)
+
+Windows:
+`python -m venv venv`
+
+Linux: `python3 -m venv venv` (untested)
```
Comment (T0mmas) [inline README.md]:
```suggestion
Linux: `python3 -m venv venv`
```

File: client.py [hunk 3]
```diff
@@ -46,14 +46,17 @@ def receive_and_decrypt_msg(self):
...
         self.noise_handshake()
-        self.send_encrypted_msg("Hello server!")
+        self.send_encrypted_msg(message)
         self.receive_and_decrypt_msg()
 
 
 if __name__ == "__main__":
     client = Client()
-    client.run()
+    if len(argv):
```
Comment (T0mmas) [inline client.py]:
```suggestion
    if len(argv) > 1:
```

Comment (Matej4545) [reply client.py]: Jej to mi nejak ulitlo :D

Comment (T0mmas) [reply client.py]: Pohoda, od toho máme ty reviews :D

**Review by T0mmas — APPROVED** [review]

**Pull request merged**